tests/datastore.spec.ts

```diff
@@ -42,221 +45,180 @@ describe('Google Cloud Storage', () => {
-          };
+      test('should add an entity', (done) => {
+        const cloudDatabase = getCloudDatabase(storageConfig);
+        // Mock the `Datastore#save()` method to complete successfully, simulating a successful addition.
+        jest.spyOn(Datastore.prototype, 'save').mockResolvedValue(undefined as never);
+        loggerMock.error.mockClear();
+
+        cloudDatabase.add('dataBasicItem1', (err: VerdaccioError) => {
+          expect(loggerMock.error).not.toHaveBeenCalled();
+          expect(err).toBeUndefined();
+
+          done();
         });
+      });
 
+      test('should handle failing to add an entity', (done) => {
         const cloudDatabase = getCloudDatabase(storageConfig);
-        cloudDatabase.add(pkgName, (err: VerdaccioError) => {
-          expect(err).not.toBeNull();
-          expect(err.code).toEqual(HTTP_STATUS.INTERNAL_ERROR);
+        // Mock the `Datastore#save()` method to fail, simulating a failed addition.
+        jest.spyOn(Datastore.prototype, 'save').mockRejectedValue(new Error('Failed to add.') as never);
+        loggerMock.error.mockClear();
+
+        cloudDatabase.add('dataBasicItem1', (err: VerdaccioError) => {
+          expect(loggerMock.error).toHaveBeenCalledWith(
+            "gcloud: [datastore add] failed to add package 'dataBasicItem1' to datastore '[default].VerdaccioPackage': Failed to add."
+          );
+
+          expect(err).toBeInstanceOf(Error);
+          expect(err.code).toBe(HTTP_STATUS.INTERNAL_ERROR);
+          expect(err.message).toBe('Failed to add.');
+
           done();
         });
       });
 
-      test('should delete an entity', done => {
-        const deleteDataStore = jest.fn();
-
-        jest.doMock('../src/storage-helper', () => {
-          const originalModule = jest.requireActual('../src/storage-helper').default;
-
-          return {
-            __esModule: true,
-            default: class Foo extends originalModule {
-              public datastore: object;
-              public constructor(props) {
-                super(props);
-                // gcloud sdk uses Symbols for metadata in entities
-                const sym = Symbol('name');
-                this.datastore = {
-                  KEY: sym,
-                  key: jest.fn(() => true),
-                  int: jest.fn(() => 1),
-                  delete: deleteDataStore,
-                  createQuery: (): string => 'query',
-                  runQuery: (): Promise<object[]> => {
-                    const entity = {
-                      name: pkgName,
-                      id: 1,
-                    };
-                    entity[sym] = entity;
-
-                    return Promise.resolve([[entity], {}]);
-                  },
-                };
-              }
-            },
-          };
+      test('should delete an entity', (done) => {
+        const cloudDatabase = getCloudDatabase(storageConfig);
+        // Mock the `Datastore#key()` method to be generate a fake key.
```

Comment:
```suggestion
        // Mock the `Datastore#key()` method to be generated a fake key.
```

Reply:
Good catch :+1:
Changed to `to generate a fake key`.